Manual Curation of Cell Type Annotations

Cell type annotations within the spatiotemporal atlas were initially assigned using a semi-automated pipeline that incorporated high-resolution clustering and label transfer from the extended gastrulation atlases. While many annotations were reliably transferred via exploration of physical localization and marker gene expression patterns, additional manual curation was performed to refine specific populations, particularly embryo proper endothelium, lateral plate mesoderm (LPM), and extraembryonic ectoderm (ExE ectoderm). Some annotations may still require further refinement, and we provide the full dataset, clustering data, and transferred labels as an open-access resource to enable researchers to make additional improvements based on their expertise.

To enhance transparency and support further evaluation, Supplementary Tables 1-4 provide detailed marker genes for each cell type, while Supplementary Videos 1-3 illustrate the spatial distribution of these annotated cell types across the seven seqFISH embryos, along with gene expression patterns for the related cell types.

Updating ExE Ectoderm and LPM Annotations
To improve the resolution of populations initially labeled as ExE ectoderm and lateral plate mesoderm (LPM) following the first round of majority voting, we refined annotations by back-referencing and assigning majority-vote seqFISH cell type labels. These labels provided crucial spatial localization information, allowing us to better resolve these populations. Subsequently, the labels were consolidated into ‘ExE ectoderm proximal’, ‘ExE ectoderm distal’, ‘ExE mesoderm and anterior LPM’, or ‘ExE mesoderm and posterior LPM’, in alignment with their distinct spatial positions as indicated by seqFISH labels.

Reclassification of Embryo Proper Endothelium
Annotations for embryo proper endothelium were refined by integrating spatial information from seqFISH embryos and clustering data. Cells initially labelled embryo proper endothelium through majority voting were reassigned to more specific subtypes, including endocardium #1/#2, Allantois endothelium, and Embryo proper endothelium #1/#2.



Refining Poorly Labeled Populations Using Spatial Information
To improve annotation accuracy, we refined a subset of primitive streak cells based on their sub-cluster identities and spatial positioning, reassigning them as rostral ectoderm. Similarly, a sub-cluster of epiblast was relabelled ExE endoderm, and another sub-cluster initially labelled haematoendothelial progenitors were reclassified as ExE mesoderm, better reflecting anatomical localization. Cells previously assigned as parietal endoderm were reassigned to visceral endoderm, while a group of anterior primitive streak cells were relabelled notochord.

Resolving Ambiguous Labels
As the initial annotation process relied on majority voting, two populations required further refinement to resolve ambiguous labels. Cells initially labelled ‘Dorsal spinal cord progenitors, Neural tube’ were relabelled as ‘Dorsal spinal cord progenitors’, while cells assigned ‘Non-neural ectoderm, Surface ectoderm’ were renamed ‘Surface ectoderm’ to reflect their spatial identity more accurately.

Overall, these refinements aimed to ensure that cell type annotations align more accurately with the spatial and transcriptomic landscape captured in our dataset. By combining automated label transfer with expert-driven curation, we improve annotation specificity while maintaining flexibility for further refinement by the research community. All updated annotations, code, and metadata are publicly available to ensure transparency and reproducibility.
